src/matrix_multiply_add.jl

```diff
@@ -0,0 +1,326 @@
+import LinearAlgebra.MulAddMul
```

Comment:
`MulAddMul` was meant to be an implementation detail of 5-arg `mul!`.  It's simple piece of code so the safest thing to do is to vendor it here, if you want to use it.  But `MulAddMul` has been causing some troubles in LinearAlgebra so I'm not entirely sure if that's the best solution.  Is just using `alpha = true` and `beta = false` as default not an option?

Reply:
Good idea about creating a copy in StaticArrays, so that we don't break compatibility with previous Julia versions. It's useful since the boolean type parameters are available at compile time so we can conditionally include or exclude terms in the generated code. Although looking at the the benchmarks there isn't much performance benefit to excluding the extra addition and multiplications, so maybe we shouldn't even bother special-casing the extra terms.

Reply:
I removed the dependency on `LinearAlgebra.MulAddMul` so it no longer breaks backwards  compatibility. I replaced it with something similar but less prone to type-instability.